server/src/domain/person/person.service.ts

```diff
@@ -117,4 +117,53 @@ export class PersonService {
 
     return true;
   }
+
+  async mergePerson(authUser: AuthUserDto, primaryPersonId: string, dto: MergePersonDto) {
+    for (const id of dto.ids) {
+      const person = await this.repository.getById(authUser.id, id);
+      if (!person) {
+        throw new BadRequestException();
+      }
+    }
+
+    const primaryPerson = await this.repository.getById(authUser.id, primaryPersonId);
+    if (!primaryPerson) {
+      throw new BadRequestException();
+    }
+
+    const mergeIds = dto.ids;
+    for (const mergePersonId of mergeIds) {
+      // Find and remove duplicated entry in asset_faces table
+      const identicalAssetIds = await this.repository.deleteIdenticalAssets(primaryPersonId, mergePersonId);
+
+      // Remove record of duplicated entry in asset_faces table
+      // and Typesense database belong to merge person
+      if (identicalAssetIds.length > 0) {
+        for (const assetId of identicalAssetIds) {
+          await this.jobRepository.queue({
+            name: JobName.SEARCH_REMOVE_FACE,
+            data: { assetId, personId: mergePersonId },
+          });
+        }
+        this.logger.debug(`Deleted ${identicalAssetIds.length} duplicated assets`);
```

Comment:
We aren't deleting assets, we're removing faces, so maybe this should say deleted 10 faces preparing for person merge.